packages/plugin-earlybirds/src/actions/earlybirdAction.ts

```diff
@@ -0,0 +1,140 @@
+    suppressInitialMessage: true,
+    validate: async () => true,
+    handler: async (runtime: IAgentRuntime, message: Memory, state: State, _options: { [key: string]: unknown }, callback: HandlerCallback) => {
+        const traceId = message.id;
+        const moxieUserInfo = state.moxieUserInfo as MoxieUser;
+        const moxieUserId = moxieUserInfo.id;
+
+        try {
+            elizaLogger.debug(traceId, `[EarlybirdAction] [${moxieUserId}] Starting Earlybird calculation`);
+            const latestMessage = message.content.text;
+            const context = composeContext({
+                state: {
+                    ...state,
+                    latestMessage: latestMessage,
+                    moxieUserId,
+                },
+                template: earlyBirdsTemplate,
+            });
+
+            const earlybirdsResponse = await await generateObjectDeprecated({
+                runtime,
+                context: context,
+                modelClass: ModelClass.MEDIUM,
+                modelConfigOptions: {
+                    modelProvider: ModelProviderName.OPENAI,
+                    temperature: 0.0,
+                    apiKey: process.env.OPENAI_API_KEY!,
+                    modelClass: ModelClass.MEDIUM
+                }
+            });
+
+            elizaLogger.debug(traceId, `[EarlybirdAction] earlybirdsResponse: ${JSON.stringify(earlybirdsResponse)}`);
+            const queryId = 5802474;
+            const opts: RunQueryArgs = {
+                queryId,
+                query_parameters: [
+                    QueryParameter.number("n_buyers", 200),
+                    QueryParameter.text("token1", earlybirdsResponse.token1.address),
+                    QueryParameter.text("token2", earlybirdsResponse.token2.address),
+                    QueryParameter.text("token3", earlybirdsResponse.token3.address),
+                    QueryParameter.text("token4", earlybirdsResponse.token4.address),
+                ]
+            }
+            const result = await client.runQuery(opts);
+            elizaLogger.debug(traceId, `[EarlybirdAction] result: ${JSON.stringify(result)}`);
+            const earlybirds = result.result.rows;
+            if (earlybirds.length === 0) {
+                await callback({
+                    text: `No earlybirds found for the tokens ${earlybirdsResponse.token1.address}, ${earlybirdsResponse.token2.address}, ${earlybirdsResponse.token3.address}, ${earlybirdsResponse.token4.address}`,
+                    action: "EARLYBIRD",
+                });
+                return true;
+            }
+            let template = earlybirdWalletAddressesTemplate;
+            template = template.replace("{{buyers}}", JSON.stringify(earlybirds));
+            const newContext = composeContext({
+                state: {
+                    ...state,
+                    latestMessage: latestMessage,
+                    moxieUserId,
+                },
```

Comment:
Seems for this template, we don't need `latestMessage` and `moxieUserId`, we can just use state in this case as it is
```suggestion
                state,
```